pkg/webhook/cert/writer/fs.go

```diff
@@ -18,147 +18,95 @@ package writer
 
 import (
 	"errors"
-	"fmt"
 	"io/ioutil"
 	"os"
 	"path"
-	"strings"
 
-	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
-	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime"
-	certgenerator "sigs.k8s.io/controller-runtime/pkg/webhook/cert/generator"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/cert/generator"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/cert/writer/internal/atomic"
 )
 
-const (
-	// FSCertProvisionAnnotationKeyPrefix should be used in an annotation in the following format:
-	// fs.certprovisioner.kubernetes.io/<webhook-name>: path/to/certs/
-	// the webhook cert manager library will provision the certificate for the webhook by
-	// storing it under the specified path.
-	// format: fs.certprovisioner.kubernetes.io/webhookName: path/to/certs/
-	FSCertProvisionAnnotationKeyPrefix = "fs.certprovisioner.kubernetes.io/"
-)
+// fsCertWriter provisions the certificate by reading and writing to the filesystem.
+type fsCertWriter struct {
+	// dnsName is the DNS name that the certificate is for.
+	dnsName string
 
-// FSCertWriter provisions the certificate by reading and writing to the filesystem.
-type FSCertWriter struct {
-	CertGenerator certgenerator.CertGenerator
+	// certGenerator generates the certificates.
+	certGenerator generator.CertGenerator
+	// path is the directory that the certificate and private key and CA certificate will be written.
+	path string
```

Comment:
Can you use `FSCertWriterOptions` instead of path and certGenerator? It has better readability